configure.d/1_bio_alloc.conf

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+#
+# Copyright(c) 2012-2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+. $(dirname $3)/conf_framework.sh
+
+check() {
+	cur_name=$(basename $2)
+	config_file_path=$1
+	if compile_module $cur_name "void *t=bio_alloc(0,0);" "linux/bio.h"
+	then
+		echo $cur_name "1" >> $config_file_path
+	elif compile_module $cur_name "void *t=bio_alloc(NULL,0,0,0);" "linux/bio.h"
+	then
+		echo $cur_name "2" >> $config_file_path
+	else
+		echo $cur_name "X" >> $config_file_path
+    fi
+}
+
+apply() {
+    case "$1" in
+    "1")
+		add_function "static inline struct bio *cas_bio_alloc(gfp_t gfp_mask, unsigned short num_vecs)
+		{
+			return bio_alloc(gfp_mask, num_vecs);
+		}" ;;
+    "2")
+		add_function "static inline struct bio *cas_bio_alloc(gfp_t gfp_mask, unsigned short num_vecs)
+		{
+			return bio_alloc(NULL, num_vecs, 0, gfp_mask);
```

Comment:
Somewhere in `submit_bio_noacct()` (which is called by `submit_bio()`) we have:
```c
struct block_device *bdev = bio->bi_bdev;
struct request_queue *q = bdev_get_queue(bdev);
```
So I suppose we want to set `bdev` to something other than `NULL`.